wizard/src/utils/heuristics.ts

```diff
@@ -14,13 +14,115 @@
  * limitations under the License.
  */
 
-import { CsvData, MappedThing, Mapping } from "../types";
+import {
+  Column,
+  ConfidenceLevel,
+  CsvData,
+  DCProperty,
+  DCType,
+  DetectedDetails,
+  MappedThing,
+  Mapping,
+  MappingType,
+  MappingVal,
+} from "../types";
+import { PlaceDetector } from "./detect_place";
+
+function countryOrder(detectedCountries: Map<number, DetectedDetails>): number {
+  // The detection order for type:Country is:
+  //  1. ISO code
+  //  2. Alpha Numberic 3 Letter Abbreviation
+  //  3. Numeric code
+  //  4. Country name.
+
+  const propDetected = new Map<string, number>();
+
+  detectedCountries.forEach((details: DetectedDetails, index: number) => {
+    const prop = details.detectedTypeProperty.dcProperty.dcid;
+    propDetected.set(prop, index);
+  });
+
+  if (propDetected.has("isoCode")) {
```

Comment:
nit: can just have a global constant holding an array with the country types ordered from most preferred to least preferred. Then, here, we can just iterate through the array & return if propDetected has that country type?

Reply:
I am not sure that will improve either the functionality or the readability? I wanted to keep all the country property preference order local to this one function so that there are no other global (or other) dependencies. Let me know if you feel strongly?

Reply:
I don't feel strongly, I just think it'll make the code cleaner by not having so many else-ifs

Reply:
Ok I made the suggested change except not by having a global array. The order array is local to the function.